### PR TITLE
Vulkan: limit device local memory usage to 80% of capacity

### PR DESCRIPTION
**Pull request by thatcosmonaut**

### CODE REVIEW

File: src/FNA3D_Driver_Vulkan.c [hunk 1]
```diff
@@ -3056,6 +3080,12 @@ static uint8_t VULKAN_INTERNAL_FindAvailableMemory(
 		allocator->nextAllocationSize = SDL_min(allocator->nextAllocationSize * 2, MAX_ALLOCATION_SIZE);
 	}
 
+	if (isDeviceLocal && renderer->deviceLocalHeapUsage + allocationSize > renderer->maxDeviceLocalHeapUsage)
```
Comment (flibitijibibo) [inline src/FNA3D_Driver_Vulkan.c:3083]:
```suggestion
	if (	isDeviceLocal &&
		(renderer->deviceLocalHeapUsage + allocationSize > renderer->maxDeviceLocalHeapUsage)	)
```

File: .gitignore [hunk 2]
```diff
@@ -10,4 +10,5 @@ Release/
 *.suo
 .DS_Store
 xcuserdata/
-*.xcworkspace/
\ No newline at end of file
+*.xcworkspace/
```
Comment (flibitijibibo) [inline .gitignore:13]: Loose gitignore stuff...

**Review by flibitijibibo — CHANGES_REQUESTED** [review]

**Review by flibitijibibo — APPROVED** [review]

**Comment by flibitijibibo** [comment]
Marking as a draft - the patch is perfectly fine as-is, but until we get defrag support we'll have to be a bit greedy with device-local memory.

**Comment by thatcosmonaut** [comment]
Superseded by https://github.com/FNA-XNA/FNA3D/pull/110

**Pull request closed without merging**